File: backend/src/main/java/be/lpbconsult/befintax/wallet/dto/TaxGainCalculationDTO.java
package be.lpbconsult.befintax.wallet.dto;

import java.math.BigDecimal;
import java.util.List;

public record TaxGainCalculationDTO(
        BigDecimal totalGrossGain,      // Gain réel économique
        BigDecimal totalTaxableGain,    // Gain après "Moment Photo"
        BigDecimal exemptionApplied,    // Montant de l'abattement utilisé
        BigDecimal finalTaxableBase,    // Montant réellement taxé (après abattement)
        BigDecimal estimatedTax,
        List<TaxTransactionDetailDTO> transactions
) {}

File: frontend/angular/src/app/features/wallet/dtos/TaxGainCalculation.dto.ts
export interface TaxTransactionDetailDto {
  date: string;
  assetName: string;
  quantity: number;
  sellPrice: number;
  fiscalCost: number;
  taxableGain: number;
}

export interface TaxGainCalculationDto {
  totalGrossGain: number;
  totalTaxableGain: number;
  exemptionApplied: number;
  finalTaxableBase: number;
  estimatedTax: number;
  transactions: TaxTransactionDetailDto[];
}
